src/request/components/RequestDetailPanelRequest.tsx

```diff
@@ -72,11 +74,34 @@ export class Request extends React.Component<IRequestProps, {}> {
         );
     }
 
-    private renderBody(body: string) {
+    private renderBody(body: string, headers: { [key: string]: string }) {
+        const contentType = this.getContentTypeFromHeaders(headers);
+        const highlightClassName = this.getHighlightClassNameForContentType(contentType);
+
         return (
             <div className='tab-request-body'>
-                <Highlight className=''>{body}</Highlight>
+                <Highlight className={highlightClassName}>{body}</Highlight>
             </div>
         );
     }
+
+    private getContentTypeFromHeaders(headers: { [key: string]: string }): string {
+        let contentType = undefined;
+
+        _.forEach(headers, (value, key) => {
+            if (key.toLowerCase() === 'content-type') {
+                contentType = value;
+                
+                return false;
+            }
+        });
+
+        return contentType;
+    }
+
+    private getHighlightClassNameForContentType(contentType: string): string {
+        const category = requestConverter.getContentTypeCategory(contentType);
+
+        return (category && category.highlight) || '';
+    }
```

Comment:
I'd agree--the use of Highlight.js is strictly an implementation detail of the view and so I'd want to limit exposure of aspects of that component (like the class name) to the view.  (Originally I had the whole mapping in this component as well, but then ran across a similar mapping in `request-converter` so just piggy-backed onto that instead.)